src/main/java/frc/robot/commands/DefaultMechCommand.java

```diff
@@ -0,0 +1,176 @@
+            return mReadyForIntake;
+        }
+        else if (!getIntakeBeamBreak() && getIndexerBeamBreak()) {
+            return mIntaken;
+        }
+        else if (!getIntakeBeamBreak() && getIndexerBeamBreak() && mAimingSubsystem.getCurrentState() == AimState.HANDOFF) {
+            return mReadyForHandoff;
+        }
+        else if (!getIndexerBeamBreak()) {
+            return mReadyForAim;
+        }
+        else if (!getIndexerBeamBreak() && isFlywheelAtSP() && isAimAtSP() && isVisionAligned() && Input.getRightBumper()) {
+            return mReadyForLaunch;
+        }
+        return mUltraInstinct;
+    }
+
+    @Override
+    public void execute() {
+        if (Input.getX()) {
+            mMechState.brakeLauncher();
+        }
+        else if (Input.getY()) {
+            mMechState.speakerPosition();
+        } 
+        else if (Input.getA()) {
+            mMechState.ampPosition();
+        }
+        else if (Input.getB()) {
+            mMechState.trapPosition();
+        }
+        if (Input.getLeftBumper()) {
+            mMechState.intakeInnerMotor();
+            mMechState.intakeOuterMotor();
+        } else{
+            mMechState.brakeIntake();
+        }
+        if (Input.getRightBumper()) {
+            mMechState.launch();
+        }
+        if (Math.abs(Input.getLeftStickY()) > 0) {
+            mMechState.controlWrist(Input.getLeftStickY()*AimingConstants.MAX_WRIST_TELEOP_INCREMENT);
+        }
+        if (Math.abs(Input.getRightStickY()) > 0) {
+            mMechState.controlElevator(Input.getLeftStickY()*AimingConstants.MAX_WRIST_TELEOP_INCREMENT);
+        }
+        if (Input.getDPad() == Input.DPADUP) {
+            mMechState.setElevatorSP(AimState.CLIMB);
+        } else if (Input.getDPad() == Input.DPADDOWN) {
+            mMechState.setElevatorSP(AimState.STOW);
+        }
+        if (Input.getDPad() == Input.DPADRIGHT){
+            mUseUltraInstinct = !mUseUltraInstinct;
+        }
+
+        runAction();
+        mMechState = determineState();
+    }
+
+    //automatic actions
+    public void runAction() {
+        if (mMechState == mReadyForIntake) {
+            mMechState.handoffPosition();
+        }
+        else if (mMechState == mIntaken) {
+            mMechState.handoffPosition();
+        }
+        else if (mMechState == mReadyForHandoff) {
+            mMechState.intakeInnerMotor();
+            mMechState.index();
+        }
+        else if (mMechState == mReadyForAim) {
+            //No Automation Yet
+        }
+        else if (mMechState == mReadyForLaunch) {
+            //Need Operator Confirmation
+        }
+    }
+
+    public boolean getIntakeBeamBreak() {
+        return mIntakeSubsystem.pieceInIntake();
+    }
+
+    public boolean getIndexerBeamBreak() {
+        return mLauncherSubsystem.pieceInIndexer();
+    }
+
+    public boolean isVisionAligned() {
+        return true; //V2: return LimelightOB.getNoteAlignmentCommand(swerve)
```

Comment:
What does this comment mean? Is it relevant?

Reply:
It just indicates what we will do in the future to populate this method. I can delete it for now since it serves no purpose currently.

Reply:
just replaced the comment with something more clear